### PR TITLE
TRUNK-5215: Replaced shouldIsValid test with multiple tests; added ch…

### PR DESCRIPTION
**Pull request by jerryb8**
Replaced shouldIsValid test with multiple tests; added character transposition test.

Added and replaced tests in VerhoeffIdentifierValidatorTest
Clarified Exception messaging in BaseHyphenatedIdentifierValidator

see https://issues.openmrs.org/browse/TRUNK-5215

### CODE REVIEW

File: api/src/test/java/org/openmrs/patient/impl/VerhoeffIdentifierValidatorTest.java [hunk 1]
```diff
@@ -33,87 +33,138 @@
...
+				}
+				// Shuffle the allowed identifier string around so that first character becomes last.
+				allowedIdentifier = allowedIdentifier.substring(1, allowedIdentifier.length()) + allowedIdentifier.charAt(0);
 			}
-			catch (Exception e) {}
-		}
-		
-		//Now test allowed identifiers that just have the wrong check digit.
-		for (int i = 0; i < allowedIdentifiers.length; i++) {
-			assertFalse(validator.isValid(allowedIdentifiers[i] + "-" + unusedCheckDigit));
```
Comment (dkayiwa) [inline api/src/test/java/org/openmrs/patient/impl/VerhoeffIdentifierValidatorTest.java]: Did you intentionally get rid of these?

File: api/src/test/java/org/openmrs/patient/impl/VerhoeffIdentifierValidatorTest.java [hunk 2]
```diff
@@ -33,87 +33,138 @@
...
-		//Make sure check digit can't be numeric
-		for (int j = 0; j < invalidIdentifiers.length; j++) {
-			try {
-				validator.isValid(allowedIdentifiers[j] + "-" + allowedIdentifiersCheckDigitsInt[j]);
-				fail("Identifier " + allowedIdentifiers[j] + " should have failed.");
+	}
+
+	@Test
+	public void checkDigit_shouldChangeWhenAdjacentCharsAreTransposed() {
+		VerhoeffIdentifierValidator v = new VerhoeffIdentifierValidator();
```
Comment (dkayiwa) [inline api/src/test/java/org/openmrs/patient/impl/VerhoeffIdentifierValidatorTest.java]: I would rename v to validator.

File: api/src/main/java/org/openmrs/patient/impl/BaseHyphenatedIdentifierValidator.java [hunk 3]
```diff
@@ -54,8 +54,15 @@ public String getValidIdentifier(String undecoratedIdentifier) throws UnallowedI
 	@Override
 	public boolean isValid(String identifier) throws UnallowedIdentifierException {
 		
-		if (identifier.indexOf("-") < 1) {
-			throw new UnallowedIdentifierException("Identifier must contain something besides the check digit.");
+		if (identifier.indexOf("-") <= 0) {
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/patient/impl/BaseHyphenatedIdentifierValidator.java]: Does <= 0 make it any better than < 1?

File: api/src/test/java/org/openmrs/patient/impl/VerhoeffIdentifierValidatorTest.java (hunk 1 above)
Comment (jerryb8) [reply api/src/test/java/org/openmrs/patient/impl/VerhoeffIdentifierValidatorTest.java]: This is covered in the test  isValid_shouldFailWithIncorrectlyCalculatedSuffixes()

File: api/src/test/java/org/openmrs/patient/impl/VerhoeffIdentifierValidatorTest.java (hunk 2 above)
Comment (jerryb8) [reply api/src/test/java/org/openmrs/patient/impl/VerhoeffIdentifierValidatorTest.java]: Agree, I will rename it.

File: api/src/main/java/org/openmrs/patient/impl/BaseHyphenatedIdentifierValidator.java (hunk 3 above)
Comment (jerryb8) [reply api/src/main/java/org/openmrs/patient/impl/BaseHyphenatedIdentifierValidator.java]: Sorry, I think I changed it to <0, then realised that it should fail on zero, so I added an "=" sign. I should have reinstated <1.

**Pull request merged**